src/components/TodoList.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useContext } from 'react';
+import { TodosContext } from './TodosContext';
+import { TodoItem } from './TodoItem';
+import { TodosFilter } from '../types/TodosFilter';
+import { Todo } from '../types/Todo';
+
+export const TodoList: React.FC = () => {
+  const context = useContext(TodosContext);
+
+  if (!context) {
+    return null;
+  }
+
+  const { todos, todosFilter } = context;
+
+  const filterTodos = (items: Todo[], filter: TodosFilter): Todo[] => {
+    return items.filter((todo) => {
+      switch (filter) {
+        case TodosFilter.active:
+          return !todo.completed;
+        case TodosFilter.completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  };
+
+  return (
+    <ul className="todo-list" data-cy="todosList">
+      {filterTodos(todos, todosFilter).map((todo) => (
```

Comment:
Consider will be better if you create a variable for this checking and use it here `filterTodos(todos, todosFilter)`
```suggestion
      {filterTodos(todos, todosFilter).map((todo) => (
```